nilearn/_utils/param_validation.py

```diff
@@ -221,6 +223,11 @@ def check_feature_screening(
             f" [0, 100], got {screening_percentile:g}"
         )
     else:
+        if isinstance(mask_img, SurfaceImage):
+            return SelectPercentile(
+                f_test, percentile=int(screening_percentile)
+            )
+
```

Comment:
I am wondering what would happen if the `mask_img` here already has fewer voxels/vertices than the `screening_percentile`.

With the nii image case, if this is the case, `adjust_screening_percentile` ensures that all the voxels are selected as features but we skip that step here...

Reply:
This means that we need a specific unit test, and possibly a fix.

Reply:
Ok so as of now, it picks `screening percentile` of the `mask_img`:

The `mask_img`:
![output](https://github.com/user-attachments/assets/69604f0a-e388-47c4-86b4-539c43f0e2ec)

The selected vertices after screening:
![selected](https://github.com/user-attachments/assets/21499d79-1328-4d7d-bd2e-5b3d21356bcd)